parser/lib/jsonapi/parser.rb

```diff
@@ -23,6 +23,7 @@ module JSONAPI
   #     objects in the primary data must have an id.
   # @return [JSONAPI::Parser::Document]
   def parse(document, options = {})
+    raise Parser::InvalidDocument, 'document cannot be nil' if document.nil?
```

Comment:
I'm thinking that we might as well go the extra mile and require `document` to be a `Hash`.